From 49fb00b1824c9223facefe2fa4463863f9b5685f Mon Sep 17 00:00:00 2001
From: Thao Luong <thao.luong.uw@rvc.renesas.com>
Date: Wed, 13 May 2015 14:53:18 +0700
Subject: [PATCH 3/3] omxaacdec: Update information that Renesas MC supports

This patch file update cap template base on information of Renesas MC
support for aacdec
sink:
stream-format= { raw, adts }
rate=[8000,96000]
channels=[1,6]

src:
audio/x-raw
rate = [ 8000, 96000 ]
channels = [1, 2]
format = S16LE

Current omxaacdec has using downmix function, so output is always 2
channels

refs #69205
Signed-off-by: Thao Luong <thao.luong.uw@rvc.renesas.com>
---
 omx/gstomxaacdec.c   | 15 +++++++++++++++
 omx/gstomxaudiodec.c |  8 ++++++++
 2 files changed, 23 insertions(+)

diff --git a/omx/gstomxaacdec.c b/omx/gstomxaacdec.c
index 7d81269..6f42214 100644
--- a/omx/gstomxaacdec.c
+++ b/omx/gstomxaacdec.c
@@ -1,5 +1,6 @@
 /*
  * Copyright (C) 2014, Sebastian Dröge <sebastian@centricular.com>
+ * Copyright (C) 2015, Renesas Electronics Corporation
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -61,6 +62,19 @@ gst_omx_aac_dec_class_init (GstOMXAACDecClass * klass)
   audiodec_class->get_channel_positions =
       GST_DEBUG_FUNCPTR (gst_omx_aac_dec_get_channel_positions);
 
+#ifdef USE_OMX_TARGET_RCAR
+  audiodec_class->cdata.default_sink_template_caps = "audio/mpeg, "
+      "mpegversion=(int){2, 4}, "
+      "stream-format=(string) { raw, adts }, "
+      "rate=(int)[8000,96000], "
+      "channels=(int)[1, 6], " "framed=(boolean) true";
+
+  gst_element_class_set_static_metadata (element_class,
+      "OpenMAX AAC Audio Decoder",
+      "Codec/Decoder/Audio",
+      "Decode AAC audio streams",
+      "Renesas Electronics Corporation");
+#else
   audiodec_class->cdata.default_sink_template_caps = "audio/mpeg, "
       "mpegversion=(int){2, 4}, "
       "stream-format=(string) { raw, adts, adif, loas }, "
@@ -72,6 +86,7 @@ gst_omx_aac_dec_class_init (GstOMXAACDecClass * klass)
       "Codec/Decoder/Audio",
       "Decode AAC audio streams",
       "Sebastian Dröge <sebastian@centricular.com>");
+#endif
 
   gst_omx_set_default_role (&audiodec_class->cdata, "audio_decoder.aac");
 }
diff --git a/omx/gstomxaudiodec.c b/omx/gstomxaudiodec.c
index e35af30..ff8ea53 100644
--- a/omx/gstomxaudiodec.c
+++ b/omx/gstomxaudiodec.c
@@ -93,11 +93,19 @@ gst_omx_audio_dec_class_init (GstOMXAudioDecClass * klass)
       GST_DEBUG_FUNCPTR (gst_omx_audio_dec_handle_frame);
 
   klass->cdata.type = GST_OMX_COMPONENT_TYPE_FILTER;
+#ifdef USE_OMX_TARGET_RCAR
+  klass->cdata.default_src_template_caps =
+      "audio/x-raw, "
+      "rate = (int) [ 8000, 96000 ], "
+      "channels = (int) [1, 2], "
+      "format = (string) S16LE";
+#else
   klass->cdata.default_src_template_caps =
       "audio/x-raw, "
       "rate = (int) [ 1, MAX ], "
       "channels = (int) [ 1, " G_STRINGIFY (OMX_AUDIO_MAXCHANNELS) " ], "
       "format = (string) " GST_AUDIO_FORMATS_ALL;
+#endif
 }
 
 static void
-- 
2.4.0

